### PR TITLE
Fixing common spelling mistakes with help of codespell

### PR DESCRIPTION
**Pull request by dietmarw**
Also manually checked for false positives afterwards.

### CODE REVIEW

File: Modelica/package.mo [hunk 1]
```diff
@@ -7937,8 +7937,8 @@ class Version_1_5 "Version 1.5 (Dec. 16, 2002)"
 <ul>
 <li>All example models made <strong>encapsulated</strong></li>
 <li>Upper case constants changed to lower case (cf. Modelica.Constants)</li>
-<li>Introduced Modelica.SIunits.Wavelength due to typo \"Wavelenght\"</li>
-<li>Introduced ModelicaAdditions.Blocks.Logical.Interfaces.Comparison due to typo \"Comparision\"</li>
+<li>Introduced Modelica.SIunits.Wavelength due to typo \"Wavelength\"</li>
+<li>Introduced ModelicaAdditions.Blocks.Logical.Interfaces.Comparison due to typo \"Comparison\"</li>
```
Comment (beutlich) [inline Modelica/package.mo]: Here the old typo is explicitly mentioned. I propose to remove them, as otherwise they are detected again and again.

Comment (dietmarw) [reply Modelica/package.mo]: Makes sense. Will push right away.

File: Modelica/Electrical/Digital.mo [hunk 2]
```diff
@@ -2251,7 +2251,7 @@ sum   <strong>Adder4</strong>.c_out  <strong>Adder4.s</strong>  <strong>Adder3.s
               fillPattern=FillPattern.Solid), Line(points={{50,0},{92,0}},
                 color={127,0,127})}),
             Documentation(info="<html>
-<p>SISO is a partial model for the connection pattern with <strong>s</strong>ingle (scalar) digital <strong>i</strong>nput and <strong>s</strong>ingle (scalar) digital <strong>o</strong>utput. Besides the connectors it provides a rectangle for the icon which can be filled in by the component which inherits the SISO model.</p>
+<p>SISO is a partial model for the connection pattern with <strong>s</strong>ingle (scalar) digital <strong>i</strong>nput and <strong>s</strong>ingle (scalar) digital <strong>o</strong>output. Besides the connectors it provides a rectangle for the icon which can be filled in by the component which inherits the SISO model.</p>
```
Comment (beutlich) [inline Modelica/Electrical/Digital.mo]: This one (and the following matches) should be reverted.

File: Modelica/Resources/ReleaseNotes/Modelica.UsersGuide.ReleaseNotes.Version_1_5.html [hunk 3]
```diff
@@ -332,10 +332,10 @@ <h2><a name="info" id="info">Information</a></h2>
 <li>Upper case constants changed to lower case (cf.
 Modelica.Constants)</li>
 <li>Introduced Modelica.SIunits.Wavelength due to typo
-"Wavelenght"</li>
+"Wavelength"</li>
 <li>Introduced
 ModelicaAdditions.Blocks.Logical.Interfaces.Comparison due to typo
-"Comparision"</li>
+"Comparison"</li>
```
Comment (beutlich) [inline Modelica/Resources/ReleaseNotes/Modelica.UsersGuide.ReleaseNotes.Version_1_5.html]: Those two matches will be fixed anyway when the release notes are regenerated with your help.

File: Modelica/Electrical/Digital.mo (hunk 2 above)
Comment (dietmarw) [reply Modelica/Electrical/Digital.mo]: Ah damn, I didn't see that. Will do

File: Modelica/Resources/ReleaseNotes/Modelica.UsersGuide.ReleaseNotes.Version_1_5.html (hunk 3 above)
Comment (dietmarw) [reply Modelica/Resources/ReleaseNotes/Modelica.UsersGuide.ReleaseNotes.Version_1_5.html]: Oh well

Comment (dietmarw) [reply Modelica/Resources/ReleaseNotes/Modelica.UsersGuide.ReleaseNotes.Version_1_5.html]: Actually I don't think we are readding the old release notes.

Comment (beutlich) [reply Modelica/Resources/ReleaseNotes/Modelica.UsersGuide.ReleaseNotes.Version_1_5.html]: Simply remove the old typo then (as you did for the source).

Comment (dietmarw) [reply Modelica/Resources/ReleaseNotes/Modelica.UsersGuide.ReleaseNotes.Version_1_5.html]: well it is now.

File: ModelicaReference/package.mo [hunk 4]
```diff
@@ -768,7 +768,7 @@ Define graphical layout of the parameter menu.
 The annotations <strong><code>tab</code></strong> and <strong><code>group</code></strong> define the placement of the component or of variables in a dialog with optional tab and group specification. If <code><strong>enable</strong> = false</code>, the input field may be disabled [and no input can be given]. If <code><strong>showStartAttribute</strong> = true</code> the dialog should allow the user to set the start-value and the fixed attribute for the variable instead of the value-attribute [this is primarily intended for non-parameter values and avoids introducing a separate parameter for the start-value of the variable].
 </p>
 <p>
-If <code>colorSelector=true</code>, it indicates that an rgb-value selector can be presented for a vector of  three elements and generate values <code>0..255</code> (the annotation should be useable both for vectors of Integers and Reals).
+If <code>colorSelector=true</code>, it indicates that an rgb-value selector can be presented for a vector of  three elements and generate values <code>0..255</code> (the annotation should be usable both for vectors of Integers and Reals).
```
Comment (beutlich) [inline ModelicaReference/package.mo]: Both spellings are legal. But fine for me.

**Comment by thorade** [comment]
Did you also run it in interactive mode?

**Comment by dietmarw** [comment]
No.

**Comment by thorade** [comment]
I only did it once, some time ago, but IIRC interactive mode will find a lot more errors.
Looking at the explanation of the dictionary format:
https://github.com/codespell-project/codespell#dictionary-format
Non-interactive will only handle case 1,
while interactive mode also goes through case 2 entries.

**Pull request merged**